### PR TITLE
update documentation about mutate

### PR DESCRIPTION
**Pull request by jan-paulus**
Includes and extends #321 (credits to @akx)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -387,6 +388,12 @@ try {
 }
 ```
 
+### Bound `mutate()`
+
+The SWR object returned by `useSWR` also contains a `mutate()` function that is pre-bound to the SWR's key.
+
+It is functionally equivalent to the global `mutate` function but does not require the `key` parameter.
+
```
Comment (pacocoursey) [inline README.md]: Maybe we can add a simple code example? Easier to parse than reading imo

```suggestion

const { mutate } = useSWR('/api/user')
```

**Comment by jan-paulus** [comment]
Added a proper code example as suggested. @pacocoursey

File: README.md [hunk 2]
```diff
@@ -341,7 +341,7 @@ function Profile () {
         // send a request to the API to update the data
         await requestUpdateUsername(newName)
         // update the local data immediately and revalidate (refetch)
-        mutate('/api/user', { ...data, name: newName })
+        mutate({ ...data, name: newName })
```
Comment (jasonlafferty) [inline README.md]: should this line have changed? as this isn't a bound `mutate`

Comment (quietshu) [reply README.md]: I think we should add a comment which mentions that this is the global `mutate`.

Comment (jan-paulus) [reply README.md]: Yes, my mistake. Will fix it.

**Comment by quietshu** [comment]
Closed in favor of #321.

**Pull request closed without merging**